cardinal/ecs/component.go

```diff
@@ -213,3 +214,108 @@ func WithDefault[T any](defaultVal T) ComponentOption[T] {
 		c.validateDefaultVal()
 	}
 }
+
+// Get returns component data from the entity.
+func GetComponent[T component.INameable](w *World, id entity.ID) (comp *T, err error) {
+	var t T
+	name := t.Name()
+	c, ok := w.nameToComponent[name]
+	if !ok {
+		return nil, errors.New("Must register component")
+	}
+	value, err := w.StoreManager().GetComponentForEntity(c, id)
+	if err != nil {
+		return comp, err
+	}
+	comp, ok = value.(*T)
+	if !ok {
+		return comp, fmt.Errorf("type assertion for component failed: %v to %v", value, c)
+	}
+	return comp, nil
+}
+
+// Set sets component data to the entity.
+func SetComponent[T component.INameable](w *World, id entity.ID, component any) error {
```

Comment:
is `component any` intentional? I think this could be `component T`?

Reply:
unintentional will change.